### PR TITLE
Don't create unbalanced trivia when moving types.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/17930

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs [hunk 1]
```diff
@@ -944,5 +944,57 @@ class Inner : IWhatever {
...
+{
+    class Program
+    {
+        static void Main()
+        {
+        }
+    }
+}
+
+#if true
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs]: You just know someone is going to complain about this, right?

File: src/Features/Core/Portable/CodeFixes/Async/AbstractAddAwaitCodeFixProvider.cs [hunk 2]
```diff
@@ -42,7 +43,7 @@ internal abstract partial class AbstractAddAwaitCodeFixProvider : AbstractAsyncC
         protected static bool TryGetTaskType(SemanticModel semanticModel, out INamedTypeSymbol taskType)
         {
             var compilation = semanticModel.Compilation;
-            taskType = compilation.GetTypeByMetadataName("System.Threading.Tasks.Task");
+            taskType = compilation.TaskType();
```
Comment (Pilchie) [inline src/Features/Core/Portable/CodeFixes/Async/AbstractAddAwaitCodeFixProvider.cs]: Not part of this change, right?

File: src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs [hunk 3]
```diff
@@ -944,5 +944,57 @@ class Inner : IWhatever {
...
+@"#if true
+public class Inner
+{
+
+}
+#endif";
+
+            await TestMoveTypeToNewFileAsync(
+                code, codeAfterMove, expectedDocumentName, destinationDocumentText, ignoreTrivia: false);
+        }
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs]: How about a test where it's actually a nested type too?

File: src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs]: Yes.  But at least it isn't broken code :)

Actually making it so we can remove these directives is non-trivial as we use the SyntaxEditor API and it has no facility for this.

File: src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs (hunk 3 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs]: Sure.

File: src/Features/Core/Portable/CodeFixes/Async/AbstractAddAwaitCodeFixProvider.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/CodeFixes/Async/AbstractAddAwaitCodeFixProvider.cs]: Correct.  I had a dirty file still in VS from another refactoring unfortunately.
